woocommerce/woocommerce-ios
# Order Details: sync missing variations in order details

Part of #2167

## Why

Before this PR, we only sync products in order details. Since some order items could be variations instead of products, these order items would never be loaded until the user goes to another screen that syncs the variations (e.g. visiting a variable product's variations page). This PR syncs product variations in order details so that variation order items in shipping label cards are synced for UI display. 

## Changes

- Added `ProductVariationAction.requestMissingVariations` for requesting missing variations in an order in `ProductVariationStore` with test cases
  - Unfortunately, I didn't find an API endpoint to request a list of variations so the store makes an API request for each order item that misses its variation and signals completion when all requests finish. Normally, an order wouldn't have a lot of order items
- Synced missing variations in `OrderDetailsViewModel` and `OrderDetailsViewController` following when `syncProducts` is called

## Testing

Prerequisites: the site has [WooCommerce Shipping & Tax plugin](https://docs.woocommerce.com/document/woocommerce-shipping-and-tax/) and there is an order with at least one non-refunded shipping label and an order with at least one refunded shipping label.

You can set up your store to use a test credit card with instructions in p91TBi-3xD-p2.

- In core, create an order with a variation. Then create a shipping label for this order
- Launch the app, log out and in to the store without going to the Products tab
- Go to the orders tab
- Tap on the order with a variation & shipping label --> after syncing, the variation order item should show its price, image (if any), and total

## Example screenshots 

before | after
-- | --
![Simulator Screen Shot - iPhone 11 - 2020-12-03 at 10 51 15](https://user-images.githubusercontent.com/1945542/100973580-c22b5e80-3575-11eb-90b4-476317c71676.png) | ![Simulator Screen Shot - iPhone 11 - 2020-12-03 at 13 55 27](https://user-images.githubusercontent.com/1945542/100973652-e5560e00-3575-11eb-9eb0-d8ebbb9b633a.png)


 
Update release notes:
